test/stats/detailedmetrics_test.go

```diff
@@ -0,0 +1,120 @@
+package stats
+
+import (
+	"github.com/envoyproxy/ratelimit/src/config"
+	ratelimit "github.com/envoyproxy/ratelimit/src/service"
+	settings "github.com/envoyproxy/ratelimit/src/settings"
+	"github.com/envoyproxy/ratelimit/src/stats"
+	mock_config "github.com/envoyproxy/ratelimit/test/mocks/config"
+	mock_limiter "github.com/envoyproxy/ratelimit/test/mocks/limiter"
+	mock_loader "github.com/envoyproxy/ratelimit/test/mocks/runtime/loader"
+	mock_snapshot "github.com/envoyproxy/ratelimit/test/mocks/runtime/snapshot"
+	"github.com/golang/mock/gomock"
+	gostats "github.com/lyft/gostats"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func commonSetup(t *testing.T) rateLimitServiceTestSuite {
+	ret := rateLimitServiceTestSuite{}
+	ret.assert = assert.New(t)
+	ret.controller = gomock.NewController(t)
+	ret.runtime = mock_loader.NewMockIFace(ret.controller)
+	ret.snapshot = mock_snapshot.NewMockIFace(ret.controller)
+	ret.cache = mock_limiter.NewMockRateLimitCache(ret.controller)
+	ret.configLoader = mock_config.NewMockRateLimitConfigLoader(ret.controller)
+	ret.config = mock_config.NewMockRateLimitConfig(ret.controller)
+	ret.store = gostats.NewStore(gostats.NewNullSink(), false)
+	sett := settings.NewSettings()
+	sett.DetailedMetrics = true
```

Comment:
Do we have a test that check that when DetailedMetrics is false the new metrics are not generated?